design/src/rfcs/rfc0006_crates_io_publishing.md

```diff
@@ -0,0 +1,279 @@
+RFC: Publishing to Crates.io
+============================
+
+> Status: RFC
+
+The AWS SDK for Rust and its supporting Smithy crates need to be published to [crates.io](https://crates.io/)
+so that customers can include them in their projects and also publish crates of their own that depend on them.
+
+This doc proposes a short-term and long-term solution for publishing to crates.io. The short-term approach is intended
+to be executed manually by a developer using scripts and an SOP no more than once per week, and should require less
+than a dev week to implement. The long-term approach is intended to be fully automated and run as frequently as
+once per day.
+
+Terminology
+-----------
+
+- **AWS SDK Crate**: A crate that provides a client for calling a given AWS service, such as `aws-sdk-s3` for calling S3.
+- **AWS Runtime Crate**: Any runtime crate that the AWS SDK generated code relies on, such as `aws-types`.
+- **Smithy Runtime Crate**: Any runtime crate that the smithy-rs generated code relies on, such as `smithy-types`.
+
+Requirements
+------------
+
+### Versioning
+
+Cargo uses [semver](https://github.com/dtolnay/semver#requirements) for versioning,
+with a `major.minor.patch-pre` format:
+- `major`: Incompatible API changes
+- `minor`: Added functionality in backwards compatible manner
+- `patch`: Backwards compatible bug fixes
+- `pre`: Pre-release version tag (omitted for normal releases)
+
+The Rust SDK will make one small adjustment to the above requirements: the `patch` version will be incremented
+for backwards compatible AWS model changes, and the `minor` version will be incremented for changes in Smithy runtime
+dependencies. For example, if the `smithy-types` crate version changes at all, then any crate that depends on it
+will have a `minor` version bump.
+
+Additionally, AWS SDK crates will maintain a consistent `major` and `minor` version number across all services.
+The latest version of `aws-sdk-s3` will always have the same `major.minor` version as the latest `aws-sdk-dynamodb`,
+for example. The `patch` version is allowed to be different between service crates.
+
+Smithy runtime crates will maintain their own versioning, and their version numbers are independent of each other.
+For example, the latest `smithy-types` could be at `0.3` while `smithy-json` is at `0.2`. The `aws-config` crate
+is an exception to this rule since customers depend on it directly. It should maintain the same `major.minor` as
+the AWS SDK crates.
+
+The `pre` version tag will be `alpha` during the Rust SDK alpha, `preview` during dev preview, and removed once the
+SDK is generally available.
+
+### Version Bumping
+
+In general, model changes will be frequent (daily) while changes to runtime crates and code generation
+will be infrequent (weekly to monthly). Model changes are supposed to _always_ be backwards compatible, so it
+makes sense to apply these to `patch` revisions as much as possible. That way, if a model change was made only
```

Comment:
I don't see why this makes sense. The argument is that it is to avoid updating all 260 crates, but then maybe service crates shouldn't have their minor versions synced?